### PR TITLE
Add broker routing config: newSegmentExpirationTimeInSeconds

### LINKED ISSUE #13683: Improve InstanceSelector routing logic for new segments
Recently, we experienced an incident in our production environment where numerous query failures occurred due to the absence of a replica group to route the queries to. We are using the `strictReplicaGroup` policy.

Upon debugging, we identified the following issue:

- A new consuming segment was added to the Ideal state at `19:32`
- Both replica servers received Helix transition message `SegmentOnlineOfflineStateModel.onBecomeConsumingFromOffline()` at 19:32
- Both replicas actually began consuming (`Created new consumer thread Thread:`) at 19:39, and the External view was updated at `19:39`.

Here's how the broker-segment-state updates occurred:

-  The segment was added to `_newSegmentStateMap` at approximately `19:33` : [Ref](https://github.com/apache/pinot/blob/00871f242fa7342e252d956b54d246d289887408/pinot-broker/src/main/java/org/apache/pinot/broker/routing/instanceselector/BaseInstanceSelector.java#L398-L417)
- By `19:38`, it was no longer considered a new segment due to this clause ([Ref](https://github.com/apache/pinot/blob/00871f242fa7342e252d956b54d246d289887408/pinot-broker/src/main/java/org/apache/pinot/broker/routing/instanceselector/InstanceSelector.java#L38-L40)) and was moved from `_newSegmentStateMap` to `_oldSegmentCandidatesMap`.
- At `19:38`, we started seeing query failures as both instances were tagged unavailable, thus making the replicas unavailable. Both instances were tagged unavailable because the External view was NULL for this "old-tagged" segment.
LOG:
```
Found unavailable instance: <SampleInstance1> in instance group: [SampleInstance1, SampleInstance2] for segment: 
SegmentName1, table: TableName1 (IS: {SampleInstance1=CONSUMING, SampleInstance2=CONSUMING}, EV: null)
```

One of the solution is:

- Make the `NEW_SEGMENT_EXPIRATION_MILLIS` parameter configurable here:
https://github.com/apache/pinot/blob/00871f242fa7342e252d956b54d246d289887408/pinot-broker/src/main/java/org/apache/pinot/broker/routing/instanceselector/InstanceSelector.java#L38-L40 

Note: The strategy discussed in #13284 would also have not helped in this scenario as both the replicas were not present in ExternalView.

### PR DESCRIPTION
**Pull request by tibrewalpratik17**
Fixes #13683 

Adds a new broker instance level config: newSegmentExpirationTimeInSeconds which will help in configuring the maximum amount of time broker will wait for new consuming segment's External-View to come up as `CONSUMING`. If this time expires, then the new segment is considered as "old" and may start marking instances as "unavailable". Previous default was 5 minutes.

### CODE REVIEW

**Comment by ankitsultana** [comment]
Can we keep this a node level config only? I am hesitant in adding another config to table config since these are exposed to users one way or another.

**Comment by tibrewalpratik17** [comment]
> Can we keep this a node level config only? I am hesitant in adding another config to table config since these are exposed to users one way or another.

Sure works! We can see in future if we really need to finetune this at table-level. Right now, node level makes sense to me as well. Updated!

File: pinot-spi/src/main/java/org/apache/pinot/spi/utils/CommonConstants.java [hunk 1]
```diff
@@ -344,6 +345,9 @@ public static class Broker {
     // precedence over "query.response.size" (i.e., "query.response.size" will be ignored).
     public static final String CONFIG_OF_MAX_SERVER_RESPONSE_SIZE_BYTES = "pinot.broker.max.server.response.size.bytes";
 
+    public static final String CONFIG_OF_NEW_SEGMENT_EXPIRATION_SECONDS = "pinot.broker.new.segment.expiration.seconds";
```
Comment (ankitsultana) [inline pinot-spi/src/main/java/org/apache/pinot/spi/utils/CommonConstants.java:348]: nit: Can you add a JavaDoc for this variable to explain what this config does?

**Review by ankitsultana — APPROVED** [review]

File: pinot-spi/src/main/java/org/apache/pinot/spi/utils/CommonConstants.java (hunk 1 above)
Comment (ankitsultana) [reply pinot-spi/src/main/java/org/apache/pinot/spi/utils/CommonConstants.java:348]: you can also instead update this https://docs.pinot.apache.org/configuration-reference/broker

Comment (tibrewalpratik17) [reply pinot-spi/src/main/java/org/apache/pinot/spi/utils/CommonConstants.java:348]: Thanks! Will update pinot-docs once merged!

**Comment by Jackie-Jiang** [comment]
Can you help update Pinot doc about this new config

**Pull request merged**